src/utils.jl

```diff
@@ -1,15 +1,23 @@
-using Distributions
-
-using Bijectors: Bijectors
 
+function pm_next!(pm, stats::NamedTuple)
+    ProgressMeter.next!(pm; showvalues=[tuple(s...) for s in pairs(stats)])
+end
 
-function rand_and_logjac(rng::Random.AbstractRNG, dist::Distribution)
-    x = rand(rng, dist)
-    return x, zero(eltype(x))
+function maybe_init_optimizer(
+    state_init::Union{Nothing, NamedTuple},
```

Comment:
Is `haskey` defined for `state_init`? 

It doesn't seem like `maybe_init_optimizer(::Nothing, ...)` is ever called in the current codebase?

Reply:
Oh, true `Nothing` is redundant. I'll fix that.

> Is haskey defined for state_init?

What do you mean by this?